packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -87,4 +112,47 @@ export class DefaultHasOneRepository<
     }
     return found[0];
   }
+
+  async delete(where?: Where<TargetEntity>, options?: Options): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return targetRepository.deleteAll(
+      constrainWhere(where, this.constraint as Where<TargetEntity>),
+      options,
+    );
+  }
+
+  async patch(
+    dataObject: DataObject<TargetEntity>,
+    options?: Options,
+  ): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return await targetRepository.updateAll(
+      dataObject,
```

Comment:
I think we need to call `constrainDataObject` here to prevent clients from overwriting properties they are not allowed to change (e.g. foreign key, etc.).

Please add a test to verify, see this HasMany test for inspiration: https://github.com/strongloop/loopback-next/blob/fed9956d760426766190b9c53cf7aaeed12d29f1/packages/repository/src/__tests__/acceptance/has-many.relation.acceptance.ts#L105-L112

Reply:
Ok

Reply:
> I think we need to call `constrainDataObject` here to prevent clients from overwriting properties they are not allowed to change (e.g. foreign key, etc.).
>
> Please add a test to verify.

☝️